src/main/java/io/cdap/plugin/google/drive/source/GoogleDriveSourceConfig.java

```diff
@@ -0,0 +1,295 @@
+    "Is not applicable for files in Google formats.")
+  @Macro
+  protected String maxPartitionSize;
+
+  @Name(BODY_FORMAT)
+  @Description("Output format for body of file. \"Bytes\" and \"String\" values are available. Default is \"Bytes\".")
+  @Macro
+  protected String bodyFormat;
+
+  @Name(DOCS_EXPORTING_FORMAT)
+  @Description("MIME type for exporting Google Documents. Default value is 'text/plain'.")
+  @Macro
+  protected String docsExportingFormat;
+
+  @Name(SHEETS_EXPORTING_FORMAT)
+  @Description("MIME type for exporting Google Spreadsheets. Default value is 'text/csv'.")
+  @Macro
+  protected String sheetsExportingFormat;
+
+  @Name(DRAWINGS_EXPORTING_FORMAT)
+  @Description("MIME type for exporting Google Drawings. Default value is 'image/svg+xml'.")
+  @Macro
+  protected String drawingsExportingFormat;
+
+  @Name(PRESENTATIONS_EXPORTING_FORMAT)
+  @Description("MIME type for exporting Google Presentations. Default value is 'text/plain'.")
+  @Macro
+  protected String presentationsExportingFormat;
+  private transient Schema schema = null;
+
+  public Schema getSchema() {
+    if (schema == null) {
+      schema = SchemaBuilder.buildSchema(getFileProperties(), getBodyFormat());
+    }
+    return schema;
+  }
+
+  public void validate(FailureCollector collector) {
+    super.validate(collector);
+
+    validateFileTypesToPull(collector);
+
+    validateBodyFormat(collector);
+
+    if (validateModificationDateRange(collector)
+      && getModificationDateRangeType().equals(ModifiedDateRangeType.CUSTOM)) {
+      if (checkPropertyIsSet(collector, startDate, START_DATE, START_DATE_LABEL)) {
+        checkPropertyIsValid(collector, ModifiedDateRangeUtils.isValidDateString(startDate), startDate, START_DATE,
+                             START_DATE_LABEL);
+      }
+      if (checkPropertyIsSet(collector, endDate, END_DATE, END_DATE_LABEL)) {
+        checkPropertyIsValid(collector, ModifiedDateRangeUtils.isValidDateString(endDate), startDate, END_DATE,
+                             END_DATE_LABEL);
+      }
+    }
+
+    validateFileProperties(collector);
+  }
+
+  @Override
+  protected GoogleDriveClient getDriveClient() {
+    return new GoogleDriveSourceClient(this);
+  }
+
+  private void validateFileTypesToPull(FailureCollector collector) {
+    if (!containsMacro(FILE_TYPES_TO_PULL)) {
+      if (!Strings.isNullOrEmpty(fileTypesToPull)) {
+        List<String> exportedTypeStrings = Arrays.asList(fileTypesToPull.split(","));
+        exportedTypeStrings.forEach(exportedTypeString -> {
+          try {
+            ExportedType.fromValue(exportedTypeString);
+          } catch (InvalidPropertyTypeException e) {
+            collector.addFailure(e.getMessage(), null).withConfigProperty(FILE_TYPES_TO_PULL);
+          }
+        });
+      }
+    }
+  }
+
+  private void validateBodyFormat(FailureCollector collector) {
+    if (checkPropertyIsSet(collector, BODY_FORMAT, bodyFormat, BODY_FORMAT_LABEL)) {
```

Comment:
`checkPropertyIsSet` checks should be removed. This is already checked by CDAP for non-nullable properties.
Also don't forget to check for macro.

Reply:
done